### ISSUE TITLE
Dead Code finder - UnusedVariable - false positive when usage is wrapped in if?

### ISSUE THREAD

**Issue by jaydiablo** [vimeo/psalm#836]
This code:

```php
<?php
  
class A {
    /**
     * @param array<string, string> $options
     */
	protected function setOptions(array $options): void
    {
        foreach ($options as $key => $value) {
            $normalized = ucfirst($key);
            $method     = 'set' . $normalized;

            if (method_exists($this, $method)) {
                $this->$method($value);
            }
        }
    }
  
    /**
     * @param array<string, string> $options
     */
	protected function setOptions2(array $options): void
    {
        foreach ($options as $key => $value) {
            $normalized = ucfirst($key);
            $method     = 'set' . $normalized;

            //if (method_exists($this, $method)) {
                $this->$method($value);
            //}
        }
    }
}
```

https://getpsalm.org/r/d94849bda5

Throws the UnusedVariable error when `--find-dead-code` is enabled.  As you can see from the sample, if the if condition that wraps the usage of `$value` is removed, the variable is seen as being used.

**Comment by muglug** [comment]
Yeah it's the use of `method_exists` that confuses Psalm.  Easy fix.

**Comment by muglug** [comment]
Fixed in 5867ef4
